satpy/tests/reader_tests/test_fci_l1c_fdhsi.py

```diff
@@ -0,0 +1,365 @@
+                    "scale_factor": 5,
+                    "add_offset": 10,
+                    "units": "mW.m-2.sr-1.(cm-1)-1",
+                    }
+                )
+        data[ch_path] = d
+        data[pos.format(ch_str, "start", "row")] = xrda(0)
+        data[pos.format(ch_str, "start", "column")] = xrda(0)
+        data[pos.format(ch_str, "end", "row")] = xrda(nrows)
+        data[pos.format(ch_str, "end", "column")] = xrda(ncols)
+        if pat.startswith("ir") or pat.startswith("wv"):
+            data.update(self._get_test_calib_for_channel_ir(chroot.format(ch_str),
+                        meas.format(ch_str)))
+        elif pat.startswith("vis") or pat.startswith("nir"):
+            data.update(self._get_test_calib_for_channel_vis(chroot.format(ch_str),
+                        meas.format(ch_str)))
+        data[shp.format(ch_str)] = (nrows, ncols)
+        return data
+
+    def _get_test_content_all_channels(self):
+        chan_patterns = {
+                "vis_{:>02d}": (4, 5, 6, 8, 9),
+                "nir_{:>02d}": (13, 16, 22),
+                "ir_{:>02d}": (38, 87, 97, 105, 123, 133),
+                "wv_{:>02d}": (63, 73),
+                }
+        data = {}
+        for pat in chan_patterns.keys():
+            for ch_num in chan_patterns[pat]:
+                data.update(self._get_test_content_for_channel(pat, ch_num))
+        return data
+
+    def _get_test_content_areadef(self):
+        data = {}
+        proc = "state/processor"
+        for (lb, no) in (
+                ("earth_equatorial_radius", 6378137),
+                ("earth_polar_radius", 6356752),
+                ("reference_altitude", 35786000),
+                ("projection_origin_longitude", 0)):
+            data[proc + "/" + lb] = xr.DataArray(no)
+        return data
+
+    def get_test_content(self, filename, filename_info, filetype_info):
+        # mock global attributes
+        # - root groups global
+        # - other groups global
+        # mock data variables
+        # mock dimensions
+        #
+        # ... but only what satpy is using ...
+
+        return {
+                **self._get_test_content_all_channels(),
+                **self._get_test_content_areadef(),
+                }
+
+class FakeNetCDF4FileHandler3(FakeNetCDF4FileHandler2):
+    """Mock bad data
+    """
+    def _get_test_calib_for_channel_ir(self, chroot, meas):
+        from netCDF4 import default_fillvals
+        v = xr.DataArray(default_fillvals["f4"])
+        data = {}
+        data[meas + "/radiance_unit_conversion_coefficient"] = v
+        data[chroot + "/central_wavelength_actual"] = v
+        data[meas + "/radiance_to_bt_conversion_coefficient_a"] = v
+        data[meas + "/radiance_to_bt_conversion_coefficient_b"] = v
+        data[meas + "/radiance_to_bt_conversion_constant_c1"] = v
+        data[meas + "/radiance_to_bt_conversion_constant_c2"] = v
+        return data
+
+class TestFCIL1CFDHSIReader(unittest.TestCase):
```

Comment:
It may be better to base this off of `object` and then do multiple inheritance for the "actual" test classes. Otherwise, this base class is also run/searched for tests.